graphql/query_resolver.go

```diff
@@ -988,6 +988,19 @@ func (r *queryResolver) Waterfall(ctx context.Context, options WaterfallOptions)
 		} else {
 			maxOrderOpt = order
 		}
+	} else if options.Date != nil {
+		date := utility.FromTimePtr(options.Date)
+		// Use the end of the provided date to find the most recent version created on or before it
+		eod := time.Date(date.Year(), date.Month(), date.Day(), 23, 59, 59, date.Nanosecond(), date.Location())
```

Comment:
I don't think it really matters, but is there a reason to use `date.Nanosecond()` (if it's provided) when the granularity of the date looks like it's only down to the day? I would have thought it would be hard-coded as 0, or maybe 1 second - 1 nanosecond.